app/src/main/java/com/asfoundation/wallet/ui/balance/TransactionDetailActivity.java

```diff
@@ -242,6 +244,25 @@ private void onDefaultWallet(Wallet wallet) {
             view -> viewModel.showMoreDetailsBds(view.getContext(), transaction));
         symbol = getString(R.string.p2p_send_currency_appc_c);
         break;
+      case VOUCHER:
+        button.setVisibility(View.INVISIBLE);
+        findViewById(R.id.details_label).setVisibility(View.INVISIBLE);
+        typeStr = R.string.transaction_type_voucher;
+        id = getString(R.string.voucher_buy_short_header) + " " + id;
+        typeIcon = R.drawable.ic_transaction_iab;
+        VoucherDetailView voucherDetails = findViewById(R.id.voucher_details);
+        voucherDetails.setVisibility(View.VISIBLE);
+        if (transaction.getMetadata() != null) {
+          Voucher voucher = transaction.getMetadata()
+              .getVoucher();
+          if (voucher != null) {
+            voucherDetails.setCode(voucher.getCode());
+            String redeemText = voucher.getRedeem()
+                .replaceFirst("^(http[s]?://www\\.|http[s]?://|www\\.)", "");
+            voucherDetails.setRedeemWebsite(redeemText, voucher.getRedeem());
+          }
+        }
```

Comment:
I don't think it's needed because it's not an expected error. It is an implementation requirement that if a voucher exists in the transaction list, it should return these details (if there's a service error, the voucher won't even come in the first place). I put these null checks to avoid lint warnings.